File: src/main/java/ua/softgroup/matrix/server/supervisor/producer/json/v2/Executor.java
package ua.softgroup.matrix.server.supervisor.producer.json.v2;

import java.time.LocalTime;

/**
 * @author Oleksandr Tyshkovets <sg.olexander@gmail.com>
 */
public class Executor {

    private Long id;
    private LocalTime start;
    private LocalTime end;
    private int dayWorkTimeSeconds;
    private int idleTimeSeconds;
    private double idlePercentage;
    private Report report;

    public Executor() {
    }

    public Executor(Long id, LocalTime start, LocalTime end, int dayWorkTimeSeconds, int idleTimeSeconds,
                    double idlePercentage, Report report) {
        this.id = id;
        this.start = start;
        this.end = end;
        this.dayWorkTimeSeconds = dayWorkTimeSeconds;
        this.idleTimeSeconds = idleTimeSeconds;
        this.idlePercentage = idlePercentage;
        this.report = report;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public LocalTime getStart() {
        return start;
    }

    public void setStart(LocalTime start) {
        this.start = start;
    }

    public LocalTime getEnd() {
        return end;
    }

    public void setEnd(LocalTime end) {
        this.end = end;
    }

    public int getDayWorkTimeSeconds() {
        return dayWorkTimeSeconds;
    }

    public void setDayWorkTimeSeconds(int dayWorkTimeSeconds) {
        this.dayWorkTimeSeconds = dayWorkTimeSeconds;
    }

    public int getIdleTimeSeconds() {
        return idleTimeSeconds;
    }

    public void setIdleTimeSeconds(int idleTimeSeconds) {
        this.idleTimeSeconds = idleTimeSeconds;
    }

    public double getIdlePercentage() {
        return idlePercentage;
    }

    public void setIdlePercentage(double idlePercentage) {
        this.idlePercentage = idlePercentage;
    }

    public Report getReport() {
        return report;
    }

    public void setReport(Report report) {
        this.report = report;
    }
}

File: src/main/scala/ua/softgroup/matrix/server/supervisor/producer/json/v2/Executor.scala
package ua.softgroup.matrix.server.supervisor.producer.json.v2

import java.time.LocalTime

import scala.beans.BeanProperty

/**
  * @author Oleksandr Tyshkovets <olexandr.tyshkovets@gmail.com>
  */
class Executor {

  @BeanProperty
  var id: Long = _

  @BeanProperty
  var start: LocalTime = _

  @BeanProperty
  var end: LocalTime = _

  @BeanProperty
  var dayWorkTimeSeconds: Int = _

  @BeanProperty
  var idleTimeSeconds: Int = _

  @BeanProperty
  var idlePercentage: Double = _

  @BeanProperty
  var report: Report = _

  def this(id: Long, start: LocalTime, end: LocalTime, dayWorkTimeSeconds: Int,
           idleTimeSeconds: Int, idlePercentage: Double, report: Report) {
    this()
    this.id = id
    this.start = start
    this.end = end
    this.dayWorkTimeSeconds = dayWorkTimeSeconds
    this.idleTimeSeconds = idleTimeSeconds
    this.idlePercentage = idlePercentage
    this.report = report
  }

  override def toString: String = {
    "Executor(id=%d, start=%s, end=%s, dayWorkTimeSeconds=%d, idleTimeSeconds=%d, idlePercentage=%f, report=%s)"
      .format(id, start, end, dayWorkTimeSeconds, idleTimeSeconds, idlePercentage, report)
  }

}
